.github/workflows/build-lint-test.yml

```diff
@@ -0,0 +1,41 @@
+# This workflow will install Python dependencies, run tests and lint on Python 3.11.
+# This initiates on every push or pull_request to develop
+
+name: Build, Lint, and Test
+
+on:
```

Comment:
I completely agree. I will add the workflow_dispatch trigger, and related to the discussion below on Python version, will add an option to select a specific Python version on this manual trigger so that specific versions can be tested if desired.